opm/common/util/numeric/cmp.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <math.h>
+#include <string.h>
+
+#include <cstddef>
+#include <vector>
+
+#include <opm/common/util/numeric/cmp.hpp>
+
+/**
+   Ahhh - the joys of comparing floating point numbers ....
+
+   http://www.cygnus-software.com/papers/comparingfloats/comparingfloats.htm
+*/
+
+
+namespace Opm {
+    namespace cmp {
+
+        bool double_equal(double value1 , double value2 , double abs_eps , double rel_eps) {
+            bool equal = true;
+            double diff = fabs(value1 - value2);
+            if (diff > abs_eps) {
+                double sum = fabs(value1) + fabs(value2);
+
+                if (diff > sum * rel_eps) {
+                    equal = false;
+                }
+            }
+            return equal;
+        }
+
+
+        bool double_equal(double value1 , double value2) {
+            return double_equal( value1 , value2 , abs_epsilon , rel_epsilon );
+        }
+
+        /*****************************************************************/
+
+        bool double_vector_equal(const std::vector<double>& v1,const std::vector<double>& v2, double abs_eps , double rel_eps) {
```

Comment:
Please add some whitespace to the parameter list, here it's probably best to put one argument per line.